packages/rulesets/generated/spectral/az-arm.js

```diff
@@ -2690,6 +2690,27 @@ const validatePatchBodyParamProperties = createRulesetFunction({
     return errors;
 });
 
+const requestBodyMustExistForPutPatch = (putPatchOperationParameters, _opts, ctx) => {
+    const errors = [];
+    const path = ctx.path;
+    const error = `The put or patch operation does not have a request body defined. This is not allowed. Please specify a request body for this operation.`;
+    const bodyParam = findBodyParam(putPatchOperationParameters);
+    if (bodyParam == undefined || bodyParam["schema"] == undefined) {
```

Comment:
Might want to check for empty objects here as well:
```suggestion
    if (bodyParam == undefined || bodyParam["schema"] == undefined || typeof bodyParam["schema"] !== 'object' || _.isEmpty(bodyParam["schema"])) {
```

Reply:
This condition - typeof bodyParam["schema"] !== 'object'  would not be valid as the value will be a string. Added the other condition for now. We can add deeper validation if required later.

Reply:
I believe it would be valid since you have `resolved: true` set for this rule. When that is set it should replace all of the `$ref` items with their actual values. However, we might already have some mechanism that checks for empty objects anyway.